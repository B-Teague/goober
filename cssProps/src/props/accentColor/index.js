//Do not modify this file.  This file is generated from ./scripts/generateProps.js
import { default as appendArgs } from '../../util/index'

/**
 * @name accentColor
 * @category CssProps Helper
 * @summary Returns css property accentColor and value as an object
 *
 * @description
 * Returns an object with the css property accentColor with its passed in values
 *
 * @param - One or more parameters that will be appended together based on a delimiter
 * @returns accentColor property as object
 */

export default function accentColor() {
    return {
        accentColor: appendArgs(arguments)
    }
}