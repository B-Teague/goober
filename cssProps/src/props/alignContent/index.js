//Do not modify this file.  This file is generated from ./scripts/generateProps.js
import { default as appendArgs } from '../appendArgs/index'

/**
 * @name alignContent
 * @category CssProps Helper
 * @summary Returns css property alignContent and value as an object
 *
 * @description
 * Returns an object with the css property alignContent with its passed in values
 *
 * @param - One or more parameters that will be appended together based on a delimiter
 * @returns alignContent property as object
 */

export default function alignContent() {
    return {
        alignContent: appendArgs(arguments)
    }
}