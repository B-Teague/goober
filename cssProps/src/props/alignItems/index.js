//Do not modify this file.  This file is generated from ./scripts/generateProps.js
import { default as appendArgs } from '../../util/index'

/**
 * @name alignItems
 * @category CssProps Helper
 * @summary Returns css property alignItems and value as an object
 *
 * @description
 * Returns an object with the css property alignItems with its passed in values
 *
 * @param - One or more parameters that will be appended together based on a delimiter
 * @returns alignItems property as object
 */

export default function alignItems() {
    return {
        alignItems: appendArgs(arguments)
    }
}