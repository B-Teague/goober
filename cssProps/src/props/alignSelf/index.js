/* Do not modify this file.  This file is generated from ./cssProps/templates/props.prop.index.mustache */
import { default as appendArgs } from '../../util/index';

/**
 * @name alignSelf
 * @category CssProps Helper
 *
 * @description
 * Returns an object with the css property alignSelf with its passed in values
 *
 * @param {string} One or more parameters that will be appended together separated by a delimiter based on the css syntax
 * @returns alignSelf property as object
 */

export default function alignSelf() {
    return {
        alignSelf: appendArgs(arguments)
    };
}
