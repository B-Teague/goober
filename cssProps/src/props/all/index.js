//Do not modify this file.  This file is generated from ./scripts/generateProps.js
import { default as appendArgs } from '../appendArgs/index'

/**
 * @name all
 * @category CssProps Helper
 * @summary Returns css property all and value as an object
 *
 * @description
 * Returns an object with the css property all with its passed in values
 *
 * @param - One or more parameters that will be appended together based on a delimiter
 * @returns all property as object
 */

export default function all() {
    return {
        all: appendArgs(arguments)
    }
}