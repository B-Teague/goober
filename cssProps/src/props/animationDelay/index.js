//Do not modify this file.  This file is generated from ./scripts/generateProps.js
import { default as appendArgs } from '../../util/index'

/**
 * @name animationDelay
 * @category CssProps Helper
 * @summary Returns css property animationDelay and value as an object
 *
 * @description
 * Returns an object with the css property animationDelay with its passed in values
 *
 * @param - One or more parameters that will be appended together based on a delimiter
 * @returns animationDelay property as object
 */

export default function animationDelay() {
    return {
        animationDelay: appendArgs(arguments)
    }
}