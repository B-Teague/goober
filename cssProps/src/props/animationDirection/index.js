//Do not modify this file.  This file is generated from ./scripts/generateProps.js
import { default as appendArgs } from '../appendArgs/index'

/**
 * @name animationDirection
 * @category CssProps Helper
 * @summary Returns css property animationDirection and value as an object
 *
 * @description
 * Returns an object with the css property animationDirection with its passed in values
 *
 * @param - One or more parameters that will be appended together based on a delimiter
 * @returns animationDirection property as object
 */

export default function animationDirection() {
    return {
        animationDirection: appendArgs(arguments)
    }
}