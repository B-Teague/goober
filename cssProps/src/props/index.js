export { default as display } from './props/display/index'