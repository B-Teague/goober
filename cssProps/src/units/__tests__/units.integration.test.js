import { h, createContext, render } from 'preact';
import { useContext } from 'preact/hooks';
import { setup, extractCss, css, styled } from 'goober';
import { minWidth, fontSize } from '../../props/index';
import { px, em } from '../index';

describe('cssUnits', () => {
    it('regular', () => {
        setup(h);

        const target = document.createElement('div');

        const cssClass = css([
            minWidth(px(10)),
            fontSize(em(20))
        ])

        render(
            <div className={cssClass}>
            </div>,
            target
        );

        expect(extractCss()).toMatchSnapshot();
        expect(target.innerHTML).toMatchSnapshot();
    });

    it('with theme', () => {
        const ThemeContext = createContext();
        const useTheme = () => useContext(ThemeContext);

        setup(h, null, useTheme);

        const target = document.createElement('div');

        const ThemeContainer = styled('div')((props) => ([
            minWidth(px(10)),
            fontSize(em(props.theme.minSize))
        ]));

        render(
            <ThemeContext.Provider value={{ minSize: 20 }}>
                <ThemeContainer/>
            </ThemeContext.Provider>,
            target
        );

        expect(extractCss()).toMatchSnapshot();
        expect(target.innerHTML).toMatchSnapshot();
    });
});
