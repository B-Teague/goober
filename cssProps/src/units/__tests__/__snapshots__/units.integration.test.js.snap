// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cssUnits regular 1`] = `" .go1669064858{min-width:10px;font-size:20em;}"`;

exports[`cssUnits regular 2`] = `"<div class=\\"go1669064858\\"></div>"`;

exports[`cssUnits with theme 1`] = `".go1669064858{min-width:10px;font-size:20em;}"`;

exports[`cssUnits with theme 2`] = `"<div class=\\"go1669064858\\"></div>"`;
